### PR TITLE
Update semgrep workflow

### PR DESCRIPTION
**Pull request by sfc-gh-hpathak**
Update semgrep workflow to a reusable workflow

### CODE REVIEW

**Comment by sfc-gh-hpathak** [comment]
@sfc-gh-igarish Could you approve this? I dont have access to add reviewers

**Review by sfc-gh-igarish — APPROVED** [review]
LGTM but some build is failing. If you think it's not related to your change then merge it.

**Comment by sfc-gh-hpathak** [comment]
@sfc-gh-igarish Thank you for the approval. And yes the errors arent due to my change. Also I cant merge as no write access. Could you please do the honors ? :)

**Pull request merged**
